src/components/Context/Context.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { ReactNode, useContext, useState } from 'react';
+import { Person } from '../../types';
+
+interface ValuesTypes {
+  peoples: Person[] | undefined;
+  setPeoples: React.Dispatch<React.SetStateAction<Person[] | undefined>>;
+  hasError: boolean;
+  setHasError: React.Dispatch<React.SetStateAction<boolean>>;
+  isLoading: boolean;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const TableContext = React.createContext<ValuesTypes | undefined>(undefined);
+
+if (!TableContext) {
+  throw new Error('err');
+}
+
+// eslint-disable-next-line max-len
+export const TableProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const [peoples, setPeoples] = useState<Person[] | undefined>(undefined);
```

Comment:
Setting undefined explicitly is a bad practice. you can use empty array here as initial value